azure-pipelines.yml

```diff
@@ -210,6 +210,29 @@ jobs:
       - publish: $(System.DefaultWorkingDirectory)/artifacts
         artifact: library-$(Agent.OS)
 
+  - job: Build_Docs
+    pool:
+      vmImage: "ubuntu-16.04"
+    variables:
+      rust_toolchain: nightly-2019-08-15
+    condition: in(variables['Build.SourceBranch'], 'refs/heads/master', 'refs/heads/staging', 'refs/heads/trying')
+    steps:
+      - checkout: self
+        submodules: true
+      - template: .azure/install-rust.yml
+      - template: .azure/install-llvm.yml
+      - template: .azure/install-sccache.yml
+      - template: .azure/install-cmake.yml
+      - bash: |
+          sudo apt-get install doxygen graphviz
+        displayName: Install doxygen
+      - bash: |
+          make docs
+        displayName: Build docs
+      - publish: $(System.DefaultWorkingDirectory)/api-docs
+        artifact: api-docs
+        displayName: Publish doc artifact
```

Comment:
Should we rephrase “Publish doc artifact”  to “Move [or Save] doc artifact”? I would reserve the word “publish” to pushing online/on `gh-pages`.

Reply:
doc Also, do we need to move the artifacts for `stating` and `trying`? I don't think so, maybe just for `master`?

Reply:
Cool. I will name it `Save doc artifact`.

Note: I have a plan for reusing the doc artifacts for creating nicer files for the C libraries :)